src/bp/lang-server/service/index.ts

```diff
@@ -3,22 +3,43 @@ import fs from 'fs'
 import _ from 'lodash'
 import lru from 'lru-cache'
 import ms from 'ms'
+import os from 'os'
 import path from 'path'
+import process from 'process'
 import { VError } from 'verror'
-
 import toolkit from '../../ml/toolkit'
 import Logger from '../../simple-logger'
-
 import { LoadedBPEModel, LoadedFastTextModel, ModelFileInfo, ModelSet } from './typing'
 
+interface RamInfos {
+  free: number
+  total: number
+  prediction: number
+}
+
 const maxAgeCacheInMS = ms('24h')
 
+const MARGIN_PRED = 0.2 // Adds some % to the predicted ram usage
+const MARGIN_LOW = 0.01 // How much % of total ram is considerer threshold
+
 // Examples:  "scope.en.300.bin" "bp.fr.150.bin"
 const FAST_TEXT_MODEL_REGEX = /^(\w+)\.(\w+)\.(\d+)\.bin$/i
 
 // Examples: "scope.en.bpe.model" "bp.en.bpe.model"
 const BPE_MODEL_REGEX = /^(\w+)\.(\w+)\.bpe\.model$/i
 
+const getModelSize = (dims: number) => {
+  return (7.009615385 * dims + 913.7019231) / 1000
+}
+
+const checkEnoughRam = (langNb: number, dims: number): RamInfos => {
```

Comment:
There is actually a private method that does something similar `_getResourceConsumptionInfo` did you check it ? can we adapt it so it suits our need ?

Reply:
It could... but it's different... It uses process.memoryUsage() to get the Resident Set allocated by node (V8 etc...) however, it cannot know memory used by bindings or other programs